### PR TITLE
Move AppDaysUsedRepository to browser-api so it can be accessed from autofill

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/0/1207179707350731/f 

### Description
Moves a class from app module to `browser-api` so it can be used in autofill survey.

### Steps to test this PR
- [ ] QA-optional

### CODE REVIEW

**Comment by CDRussell** [comment]
* **#4476** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4476?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4475** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4475?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4474** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4474?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CDRussell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: browser-api/src/main/java/com/duckduckgo/app/usage/app/AppDaysUsedRepository.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+package com.duckduckgo.app.usage.app
+
+import java.util.*
+
+/**
+ * Repository for storing and retrieving the number of days the app has been used
+ */
+interface AppDaysUsedRepository {
```
Comment (cmonfortep) [inline browser-api/src/main/java/com/duckduckgo/app/usage/app/AppDaysUsedRepository.kt:24]: QQ: Are we planning to revert this once survey completed?

sgtm now to unblock and release the autofill survey. However, if final, we should discuss the api change to validate what makes sense to expose, how, where, etc.

**Review by cmonfortep — APPROVED** [review]
LGTM

File: browser-api/src/main/java/com/duckduckgo/app/usage/app/AppDaysUsedRepository.kt (hunk 1 above)
Comment (CDRussell) [reply browser-api/src/main/java/com/duckduckgo/app/usage/app/AppDaysUsedRepository.kt:24]: Wasn't planning to revert, as the changes seem generally useful.

>we should discuss the api change

that was why it was added to the API proposal a few weeks back. 

>sgtm now to unblock and release the autofill survey

Ok, I will carry on with this as planned and can discuss if there should be additional changes to the API made afterwards.

**Comment by CDRussell** [comment]
### Merge activity

* **May 9, 5:30 PM EDT**: @CDRussell started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4474).

**Pull request merged**